packages/smithy-client/src/extensions/retry.ts

```diff
@@ -0,0 +1,27 @@
+import { Provider, RetryStrategy, RetryStrategyConfiguration, RetryStrategyV2 } from "@smithy/types";
+
+/**
+ * @internal
+ */
+export const getRetryConfiguration = (
+  runtimeConfig: Partial<{ retryStrategy: Provider<RetryStrategyV2 | RetryStrategy> }>
```

Comment:
Is there a reason why the type signature here is different than `resolveRetryRuntimeConfig()`?

I think we should extract the `runtimeConfig` type to it's own interface to keep it consistent.

```typescript
/**
 * @internal
 */
export interface RetryConfigurationRuntimeConfig 
  extends Partial<{ retryStrategy: Provider<RetryStrategyV2 | RetryStrategy> }> {}
```

Although not in scope of this PR, the checksum configuration should also be changed to match this pattern.